### PR TITLE
fix/add replicator counters

### PR DESCRIPTION
**Pull request by datoug**
add some new counters in replicator, also fix existing gauge counters

### CODE REVIEW

File: common/metrics/defs.go [hunk 1]
```diff
@@ -902,8 +902,10 @@ const (
 
 	// -- Replicator metrics -- //
 
-	// ReplicatorWsFailure indicates websocket failure
-	ReplicatorWsFailure
+	// ReplicatorWsFailure indicates failure when creating in stream
```
Comment (venkat1109) [inline common/metrics/defs.go]: fix comment

File: common/metrics/defs.go [hunk 2]
```diff
@@ -902,8 +902,10 @@ const (
...
+	ReplicatorCreateInStreamFailure
+	// ReplicatorWsFailure indicates failure when creating out stream
```
Comment (venkat1109) [inline common/metrics/defs.go]: fix comment

File: common/metrics/defs.go [hunk 3]
```diff
@@ -1070,21 +1072,22 @@ var metricDefs = map[ServiceIdx]map[int]metricDefinition{
...
 		ReplicatorOutConnCreditsSent:                    {Counter, "replicator.outconn.creditssent"},
 		ReplicatorOutConnMsgRead:                        {Counter, "replicator.outconn.msgread"},
-		ReplicatorReconcileDestRun:                      {Counter, "replicator.reconcile.dest.run"},
-		ReplicatorReconcileDestFail:                     {Counter, "replicator.reconcile.dest.fail"},
-		ReplicatorReconcileDestFoundMissing:             {Counter, "replicator.reconcile.dest.foundmissing"},
-		ReplicatorReconcileDestExtentRun:                {Counter, "replicator.reconcile.destextent.run"},
-		ReplicatorReconcileDestExtentFail:               {Counter, "replicator.reconcile.destextent.fail"},
-		ReplicatorReconcileDestExtentFoundMissing:       {Counter, "replicator.reconcile.destextent.foundmissing"},
-		ReplicatorReconcileDestExtentInconsistentStatus: {Counter, "replicator.reconcile.destextent.inconsistentstatus"},
+		ReplicatorReconcileDestRun:                      {Gauge, "replicator.reconcile.dest.run"},
```
Comment (venkat1109) [inline common/metrics/defs.go]: Don't you want to also update the callers to use UpdateGauge() instead of IncCounter() now ?

Comment (datoug) [inline common/metrics/defs.go]: caller is already using UpdateGauge, so this is a bug fix.

**Pull request merged**